### PR TITLE
BHBC-1604

### PR DESCRIPTION
**Pull request by KjartanE**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1591
- https://quartech.atlassian.net/browse/BHBC-1604

## This PR contains the following changes

- {List all the relevant changes. If there are many changes across many Jira tickets, organize the changes by Jira ticket}

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [x] My code follows the established style conventions
- [x] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/queries/project/project-delete-queries.ts [hunk 1]
```diff
@@ -263,6 +263,27 @@ export const deleteProjectRegionSQL = (projectId: number | undefined): SQLStatem
   return sqlStatement;
 };
 
+/**
+ * SQL query to delete range associated to a project ID.
+ *
+ * @param {number} projectId
+ * @returns {SQLStatement} sql query object
+ */
+export const deleteProjectRangeSQL = (projectId: number | undefined): SQLStatement | null => {
```
Comment (NickPhura) [inline api/src/queries/project/project-delete-queries.ts]: Do we actually expect projectId to be undefined sometimes?
We can probably just remove the `undefined` option, and then the if check below can be removed.

**Comment by NickPhura** [comment]
Don't forget to modify the yup schema for the ranges field, so that it is only required if the user specifies a caribou field. Can help with this if needed.

File: api/src/paths/project/{projectId}/update.ts [hunk 2]
```diff
@@ -249,15 +249,25 @@ PUT.apiDoc = {
               }
             },
             location: {
-              description: 'The project location object',
+              title: 'Location',
               type: 'object',
-              required: ['geometry'],
```
Comment (NickPhura) [inline api/src/paths/project/{projectId}/update.ts:258]: I think geometry is still required?

Comment (NickPhura) [reply api/src/paths/project/{projectId}/update.ts:258]: should double check the others too, especially if any of them are required by the Database.

File: api/src/queries/project/project-delete-queries.ts [hunk 3]
```diff
@@ -249,6 +249,12 @@ export const deleteProjectSpatialSQL = (projectId: number | undefined): SQLState
  * @returns {SQLStatement} sql query object
  */
 export const deleteProjectRegionSQL = (projectId: number | undefined): SQLStatement | null => {
```
Comment (NickPhura) [inline api/src/queries/project/project-delete-queries.ts]: just double checking: do we really ever send this undefined? Would be better if we can make the parent check this and then this function can just specify a number, and not worry about null checks.

File: api/src/paths/project/{projectId}/update.ts (hunk 2 above)
Comment (anissa-agahchen) [reply api/src/paths/project/{projectId}/update.ts:258]: The database doesn't require it, but our front-end does.

**Review by anissa-agahchen — DISMISSED** [review]
On create, I get a 500 if I pick a non-caribou species .... essentially the range is an empty string.

**Review by NickPhura — APPROVED** [review]

**Pull request merged**